RedBookPlayer/PlayableDisc.cs

```diff
@@ -0,0 +1,662 @@
+                // Unset image means we can't do anything
+                if(_image == null)
+                    return;
+
+                // Cache the current track for easy access
+                Track track = _image.Tracks[CurrentTrackNumber];
+
+                // Ensure that the value is valid, wrapping around if necessary
+                if(value > track.Indexes.Keys.Max())
+                    _currentTrackIndex = 0;
+                else if(value < 0)
+                    _currentTrackIndex = track.Indexes.Keys.Max();
+                else
+                    _currentTrackIndex = value;
+
+                // Set new index-specific data
+                SectionStartSector = (ulong)track.Indexes[CurrentTrackIndex];
+                TotalTime = track.TrackEndSector - track.TrackStartSector;
+            }
+        }
+
+        /// <summary>
+        /// Current sector number
+        /// </summary>
+        public ulong CurrentSector
+        {
+            get => _currentSector;
+            set
+            {
+                // Unset image means we can't do anything
+                if(_image == null)
+                    return;
+
+                // Cache the current track for easy access
+                Track track = _image.Tracks[CurrentTrackNumber];
+
+                _currentSector = value;
+
+                if((CurrentTrackNumber < _image.Tracks.Count - 1 && CurrentSector >= _image.Tracks[CurrentTrackNumber + 1].TrackStartSector)
+                        || (CurrentTrackNumber > 0 && CurrentSector < track.TrackStartSector))
+                {
+                    foreach(Track trackData in _image.Tracks.ToArray().Reverse())
+                    {
+                        if(CurrentSector >= trackData.TrackStartSector)
+                        {
+                            CurrentTrackNumber = (int)trackData.TrackSequence - 1;
+                            break;
+                        }
+                    }
+                }
+
+                foreach((ushort key, int i) in track.Indexes.Reverse())
+                {
+                    if((int)CurrentSector >= i)
+                    {
+                        CurrentTrackIndex = key;
+                        return;
+                    }
+                }
+
+                CurrentTrackIndex = 0;
+            }
+        }
+
+        /// <summary>
+        /// Represents the pre-emphasis flag
+        /// </summary>
+        public bool TrackHasEmphasis { get; private set; } = false;
+
+        /// <summary>
+        /// Indicates if de-emphasis should be applied
+        /// </summary>
+        public bool ApplyDeEmphasis { get; private set; } = false;
+
+        /// <summary>
+        /// Represents the copy allowed flag
+        /// </summary>
+        public bool CopyAllowed { get; private set; } = false;
+
+        /// <summary>
+        /// Represents the track type
+        /// </summary>
+        public TrackType? TrackType { get; private set; }
```

Comment:
Nullable?

Reply:
This mattered more in a previous iteration. I don't think that applies anymore.